bin/filecheck.py

```diff
@@ -308,13 +324,12 @@ def text(self):
 
     def application(self):
         """Process an application specific file according to its subtype."""
-        for subtype, method in self.app_subtype_methods.items():
-            if subtype in self.sub_type:
-                # TODO: should we change the logic so we don't iterate through all of the subtype methods?
-                # TODO: should these methods return a value?
-                method()
-                return
-        self._unknown_app()
+        if self.subtype in self.app_subtype_methods:
+            method = self.app_subtype_methods[self.subtype]
+            method()
+            # TODO: should these application methods return a value?
```

Comment:
I've been thinking about making the error handling better, this is a good place do it. But I don't think we need a return value. I was thinking about making the core loop of the program more functional, but it doesn't make sense for what we're doing (since we almost always take no action after reaching the end of the call graph, other than copying).